card/mod/01_core/set/all/name.rb

```diff
@@ -31,10 +31,14 @@ def name= newname
 
   newkey = cardname.key
   if key != newkey
+    was_in_cache = Card.cache.soft.delete key
     self.key = newkey
+    # keep the soft cache up-to-date
+    Card.write_to_soft_cache self if was_in_cache
```

Comment:
I'd like to see this handled in `#key=`.  Something like:

````
def key= newkey
  was_in_cache = Card.cache.soft.delete key
  write_attribute :key, newkey
  # keep the soft cache up-to-date
  Card.write_to_soft_cache self if was_in_cache
  newkey
end
````

This way the cache is handled properly if the key is set directly (though that's generally a bad idea) and `name=` is less crazy.